compiler/src/iree/compiler/Dialect/VMVX/Transforms/ResolveBufferDescriptors.cpp

```diff
@@ -316,45 +356,37 @@ struct ResolveExtractMetadataFromHalInterfaceBindingSubspan
     if (memRefType.getRank() < 1) return failure();
 
     auto loc = op.getLoc();
-    int rank = memRefType.getRank();
-    DescriptorInfo resultDescriptor;
-
-    // Compute sizes.
-    auto dynamicDimIt = binding.getDynamicDims().begin();
-    for (int i = 0; i < rank; ++i) {
-      if (memRefType.isDynamicDim(i)) {
-        resultDescriptor.sizes.push_back(*dynamicDimIt);
-        dynamicDimIt++;
-      } else {
-        resultDescriptor.sizes.push_back(
-            rewriter.getIndexAttr(memRefType.getDimSize(i)));
-      }
+    FailureOr<DescriptorInfo> resultDescriptor =
+        resolveBufferDescriptorForInterfaceBinding(binding, rewriter, loc);
+    if (failed(resultDescriptor)) {
+      return rewriter.notifyMatchFailure(
+          op, "failed to resolve descriptor with source being binding op");
     }
 
-    // Strides.
-    resultDescriptor.strides =
-        getStridesFromSizes(rewriter, loc, resultDescriptor.sizes);
-    resultDescriptor.offset = rewriter.getIndexAttr(0);
-
-    replaceOffsetSizesAndStridesWith(rewriter, op, resultDescriptor);
+    replaceOffsetSizesAndStridesWith(rewriter, op, resultDescriptor.value());
 
     // Base buffer. Use a 1D memref for hal.interface.binding.subspan.
     AffineMap mulMap = getMulMap(rewriter.getContext());
     OpFoldResult linearizedMemrefSize = rewriter.getIndexAttr(1);
-    for (auto size : resultDescriptor.sizes) {
+    for (auto size : resultDescriptor->sizes) {
       linearizedMemrefSize = makeComposedFoldedAffineApply(
           rewriter, loc, mulMap, {linearizedMemrefSize, size});
     }
     SmallVector<int64_t> staticLinearShape;
     SmallVector<Value> dynamicLinearShape;
     dispatchIndexOpFoldResult(linearizedMemrefSize, dynamicLinearShape,
                               staticLinearShape);
+    OpBuilder::InsertionGuard g(rewriter);
+    rewriter.setInsertionPoint(binding);
+
+    IntegerAttr alignment = getUpdatedAlignment(
```

Comment:
@benvanik I am rethinking this. I change the alignment here to use the alignment of the new subspan op to `MinAlign(offset, base_alignment)`. Rethinking this, is the interpretation here is that the alignment remains the same?